File: 8-kyu-kata/convert-a-number-to-a-string/javascript/convertANumberToAString.Test.js
// 8 kyu - Convert a Number to a String!  [ ID: 5265326f5fda8eb1160004c8  (convert-a-number-to-a-string) ]
// URL: https://www.codewars.com/kata/5265326f5fda8eb1160004c8
// Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
// *****************************************************************************
const assert = require('chai').assert;
describe("Tests", () => {
  it("fixed tests", () => {
    assert.strictEqual(numberToString(67), '67');
    assert.strictEqual(numberToString(79585), '79585');
    assert.notStrictEqual(numberToString(79585), 79585);
    assert.strictEqual(numberToString(1+2), '3');
    assert.strictEqual(numberToString(1-2), '-1');
  });
  
  it("random tests", () => {
    for (let run = 0; run < 200; ++run) {
      let num = Math.floor(Math.random() * 2000000 - 1000000);
      assert.strictEqual(numberToString(num), String(num), `Testing for num = ${num}`);
    }
  });
});

File: kata-8-kyu/convert-a-number-to-a-string/python/convert_a_number_to_a_string_test.py
# + =====================================================================================================================
# +
# + 8 kyu - Convert a Number to a String!  [ ID: 5265326f5fda8eb1160004c8 ] (convert-a-number-to-a-string)
# + URL: https://www.codewars.com/kata/5265326f5fda8eb1160004c8
# + Category: REFERENCE  |  Tags: STRINGS | FUNDAMENTALS
# +
# + =====================================================================================================================

import codewars_test as test
from convert_a_number_to_a_string import number_to_string


@test.describe("Fixed Tests")
def basic_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(number_to_string(67), "67")
        test.assert_equals(number_to_string(79585), "79585")
        test.assert_equals(number_to_string(79585), "79585")
        test.assert_equals(number_to_string(1 + 2), "3")
        test.assert_equals(number_to_string(1 - 2), "-1")


@test.describe("Random Tests")
def random_tests():

    from random import randint

    sol = lambda n: str(n)

    for _ in range(100):

        num = randint(1, 1e6)

        @test.it(f"Testing for number_to_string({num})")
        def test_case():
            test.assert_equals(number_to_string(num), sol(num), "It should work for random inputs too")
